stdlib/list.ml

```diff
@@ -87,6 +87,13 @@ let rec flatten = function
 
 let concat = flatten
 
+let iota ?(start = 0) ?(step = 1) len =
+  if len < 0 then invalid_arg "List.iota" else
+    let rec loop n l =
+      if n = len then rev l else
```

Comment:
Could you avoid the `rev` (and hence the double traversal) by counting down from `n` to `0` (or from `n-1` to `-1`) instead of up from `0` to `n`?

Reply:
> Could you avoid the `rev` (and hence the double traversal) by counting down from `n` to `0` (or from `n-1` to `-1`) instead of up from `0` to `n`?

I'm working on it; I still have a very slow workflow...